src/dft/backends/cufft/commit.cpp

```diff
@@ -323,6 +350,63 @@ class cufft_commit final : public dft::detail::commit_impl<prec, dom> {
         return offsets;
     }
 
+    virtual void set_workspace(scalar_type* usm_workspace) override {
+        this->external_workspace_helper_.set_workspace_metadata(*this, usm_workspace);
+        if (plans[0]) {
+            cufftSetWorkArea(*plans[0], usm_workspace);
+        }
+        if (plans[1]) {
+            cufftSetWorkArea(*plans[1], usm_workspace);
+        }
+    }
+
+    void set_buffer_workspace(cufftHandle plan, sycl::buffer<scalar_type>& buffer_workspace) {
```

Comment:
Why do we need this function if we are going to set the workspace with each compute call using buffers?

Reply:
I give my reasoning in the comment starting on line 364 - to allow internal memory to be freed (if required), and to allow cuFFT to generate any errors (I think its preferable to get errors for setting the workspace at this point in comparison to later when compute calls are made). I'm not sure what errors are likely. I'd guess the most likely is that the buffer can't be accessed on the Nvidia device.

Reply:
Sorry, yes you did write that.